source/dofs/dof_handler_policy.cc

```diff
@@ -229,6 +229,22 @@ namespace internal
       {
         /* -------------- distribute_dofs functionality ------------- */
 
+        template <int dim, int spacedim>
+        inline types::global_dof_index
+        distribute_non_local_dofs_on_cell(
+          const DoFHandler<dim, spacedim> &dof_handler,
+          const typename DoFHandler<dim, spacedim>::active_cell_iterator &cell,
+          types::global_dof_index next_free_dof)
+        {
+          const auto non_local_ids =
+            dof_handler.get_fe().get_non_local_dofs_on_cell(cell);
```

Comment:
Shouldn't this be
```suggestion
            dof_handler.get_fe().get_non_local_dofs_per_cell(cell);
```
??

Reply:
Although, actually, I'm now quite confused. Presumably, you mean for the function to return an array. What's in this array?

Reply:
This returns an `IndexSet`. Since only the finite element can know what global dofs are non-zero on this cell, we ask the finite element.